kyo-prelude/shared/src/main/scala/kyo2/kernel/Safepoint.scala

```diff
@@ -0,0 +1,121 @@
+package kyo2.kernel
+
+import internal.*
+import java.util.Arrays
+import kyo2.isNull
+import scala.annotation.tailrec
+import scala.util.control.NonFatal
+
+final class Safepoint(initDepth: Int, initState: Safepoint.State):
+    import Safepoint.State
+
+    private val owner         = Thread.currentThread()
```

Comment:
Is there any case where this value could be shared to avoid the method invocation?

```
final class Safepoint(initDepth: Int, initState: Safepoint.State, private val owner = Thread.currentThread())
```

Not sure if it's worth the complexity

Reply:
yeah, I think it should have a similar execution and only move `Thread.currentThread()` from the constructor to a default param.